pypsa/optimization/optimize.py

```diff
@@ -152,7 +154,7 @@ def define_objective(n, sns):
     m.objective = merge(objective)
 
 
-def create_model(n, snapshots=None, multi_investment_periods=False, **kwargs):
+def create_model(n, snapshots=None, multi_investment_periods=False, transmission_losses=0, **kwargs):
```

Comment:
Just an idea: what do you think about instead of adding it to the core function to define a abstract optimization in `optimization.abstract.py` like

```
def optimize_with_transmission_losses(
        n,
        snapshots=None,
        transmission_losses=0,
        multi_investment_periods=False,
        model_kwargs={},
        **kwargs,
    ):

    m = n.optimize.create_model(
        snapshots=snapshots,
        multi_investment_periods=multi_investment_periods,
        **model_kwargs,
    )

    for c in n.passive_branch_components:
        define_loss_variables(n, sns, c)
        define_loss_constraints(n, sns, c, transmission_losses):

        # modify line capacity constraint:
        for bound, kind in product(("lower", "upper"), ("fix", "ext")):
            
            vars = n.model.constraints['{c}-{kind}-s-{bound}']
            loss = reindex(n.model[f"{c}-loss"], c, fix_i)
            if bound == "lower":
                vars.lhs = vars.lhs - loss
            else:
                vars.lhs = vars.lhs + loss
                
        # and modify nodal balance constraint
        
        ...
        
```

(This is just from scratch and untested)